openmeter/ent/schema/subscription_entitlement.go

```diff
@@ -0,0 +1,47 @@
+package schema
+
+import (
+	"entgo.io/ent"
+	"entgo.io/ent/schema/edge"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+
+	"github.com/openmeterio/openmeter/pkg/framework/entutils"
+)
+
+type SubscriptionEntitlement struct {
+	ent.Schema
+}
+
+func (SubscriptionEntitlement) Mixin() []ent.Mixin {
+	return []ent.Mixin{
+		entutils.IDMixin{},
+		entutils.NamespaceMixin{},
+		entutils.TimeMixin{},
+	}
+}
+
+func (SubscriptionEntitlement) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("entitlement_id").NotEmpty().Immutable(),
+		field.String("subscription_id").NotEmpty().Immutable(),
+		field.String("subscription_phase_key").NotEmpty().Immutable(),
+		field.String("subscription_item_key").NotEmpty().Immutable(),
+	}
+}
+
+func (SubscriptionEntitlement) Indexes() []ent.Index {
+	return []ent.Index{
+		index.Fields("namespace", "id"),
+		index.Fields("namespace", "entitlement_id"),
+		index.Fields("namespace", "subscription_id"),
+		index.Fields("namespace", "subscription_id", "subscription_phase_key", "subscription_item_key"),
```

Comment:
if I'm not mistaken, PG can use partial indexes and may above not require.
But I'm fine optimizing indexes later.

Reply:
I'm not sure what you mean here. I think "Partial Index" officially refers to indexes with where clauses pretty much. If you mean why all the prefixing and not just have separate indexes, it can use multiple indexes

I didn't put too much thought into this, just added the exact indexes for the queries that will be run. I don't have a preference here at this point.